control-plane/pkg/apis/messaging/v1beta1/kafka_channel_validation.go

```diff
@@ -0,0 +1,65 @@
+/*
+ * Copyright 2020 The Knative Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package v1beta1
+
+import (
+	"context"
+	"fmt"
+
+	"knative.dev/eventing/pkg/apis/eventing"
+	"knative.dev/pkg/apis"
+)
+
+func (c *KafkaChannel) Validate(ctx context.Context) *apis.FieldError {
+	errs := c.Spec.Validate(ctx).ViaField("spec")
+
+	// Validate annotations
+	if c.Annotations != nil {
```

Comment:
we should follow the chain pattern that the `FieldError` was designed for. Basically the idea is to yank out all validation logic into functions under the top level, non-versioned directory (e.g. `knative.dev/eventing-kafka/pkg/apis/messaging/` ) and use the chain pattern syntactic sugar to call those in the correct sequence. That way you decouple the validation logic from the API shape, and you reduce code duplication in case you want to change the validation code across all the API versions. 

An example of that can be found in [serving autoscaling](https://github.com/knative/serving/blob/main/pkg/apis/autoscaling/v1alpha1/pa_validation.go#L34) for example.

Reply:
That's one way of validating things. ...a complex way 😄 

Uhm, sorry, I have been saying no to many of your comments, but would you like to send a PR for changing this? I want to merge, and do not spend more time on improving the non-functional, syntactic aspects of the validation.
That's mostly independent work, that anybody can do anyway.

Reply:
Writing this here, based on your comment:

> I consider my current review as an audit sort of review and will keep leaving the comments for reference of what needs to be done.